lib/Linter.coffee

```diff
@@ -0,0 +1,55 @@
+{Task} = require('atom')
+
+workers = new Set
+workerFile = require.resolve('./core.coffee')
+
+module.exports =
+class Linter
+
+  name: 'CoffeeLint'
+  grammarScopes: [
+    'source.coffee'
+    'source.litcoffee'
+    'source.coffee.jsx'
+    'source.coffee.angular'
+  ]
+  scope: 'file'
+  lintsOnChange: true
+
+  destroy: ->
+    worker?.terminate() for worker in workers
+
+  lint: (textEditor) ->
+    filePath = textEditor.getPath()
+    if filePath
```

Comment:
You need an `else` here returning `[]` (to clear messages) or `null` (to tell Linter not to update any messages it _might_ have for the file), otherwise you will return `undefined` and trigger a Linter message to the user that an invalid result was returned.